### ISSUE TITLE
Publish binaries for FreeBSD

### ISSUE THREAD

**Issue by zanieb** [astral-sh/uv#3370]
e.g. it currently fails with

```
# curl -LsSf https://astral.sh/uv/install.sh | sh
ERROR: there isn't a package for x86_64-unknown-freebsd

**Comment by charliermarsh** [comment]
For what it's worth, you can't upload FreeBSD wheels to PyPI, so users will always need to build any dependencies from source (IIUC).

**Comment by zanieb** [comment]
I'm a little confused since I thought we didn't have Python dependencies but that's good to know.

**Comment by charliermarsh** [comment]
Ah, to clarify, we don’t have any Python deps, I just mean: for any dependencies they install with pip or uv, IIUC they will end up building from source.

**Comment by konstin** [comment]
Note that freebsd is not supported by github actions, we either need a custom cross compiling solution or use cirrus ci

**Comment by Kartatz** [comment]
> Note that freebsd is not supported by github actions, we either need a custom cross compiling solution or use cirrus ci

Some time ago, I set up a cross-compiler targeting FreeBSD. I have not yet tried cross-compiling Rust programs myself, but that should be pretty straightforward to get working.

Here is the repository in case anyone is interested: https://github.com/AmanoTeam/Loki

**Comment by dbohdan** [comment]
https://github.com/cross-platform-actions/ lets you use Free/Net/OpenBSD in GitHub Actions. I had a good experience with it in a C project. While I haven't used it, there is also https://github.com/houseabsolute/actions-rust-cross that cross-compiles Rust.

**Comment by zanieb** [comment]
(We're currently at zero 👍 for this issue, so in addition to this being technically challenging to maintain this is pretty low priority since there's not a lot of clear interest)

**Comment by donnex** [comment]
I ran into this issue today. The `uv` package in FreeBSD ports is really old and outdated.

Has anyone been able to successfully compile the latest release of `uv` in FreeBSD? I had no luck with manual buld by cloning the repo.

**Comment by konstin** [comment]
Can you share a log of what failed in the manual build?

**Comment by donnex** [comment]
Yeah of course. I'm not sure if it's anything related to `uv` directly.

```
error: failed to run custom build command for `tikv-jemalloc-sys v0.6.0+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7`
```
[uv-cargo-build.log](https://github.com/user-attachments/files/17178728/uv-cargo-build.log)

**Comment by konstin** [comment]
Could you try https://github.com/astral-sh/uv/pull/7780?

**Comment by donnex** [comment]
@konstin works prefectly! :partying_face: :pray: 

```
target/release/uv version
uv 0.4.17 (c2f19dfd6 2024-09-29)
```

**Comment by ianthetechie** [comment]
> Note that freebsd is not supported by github actions, we either need a custom cross compiling solution or use cirrus ci

@konstin Yeah, this is really unfortunate. However, there are ways to do this while staying within the GitHub Actions ecosystem! See what I've done for a crate I maintain here: https://github.com/stadiamaps/sdf_font_tools/commit/ef7e382eda7e6f4de2b09a24387575b70c761060

**Comment by whoschek** [comment]
> Note that freebsd is not supported by github actions, we either need a custom cross compiling solution or use cirrus ci

I've been using this GitHub Action for years to run CI jobs on FreeBSD, and it is stable, fast, easy to use, and works without any issues:
https://github.com/vmactions/freebsd-vm

There is no technical problem here. 
@charliermarsh It's more a matter of uv finally getting its cross-platform priority act together.
